### ISSUE TITLE
Dockingstation no longer shown

### ISSUE THREAD

**Issue by mouse256** [kgshank/gse-sound-output-device-chooser#140]
Since the last update, the input/output of my dockinstation is no longer displayed. I've seen there's a few of changes in that context, so probably a regression.
I believe I'm running the latest version from extensions.gnome.org (not sure how exactly I can check the version?)

I still see those outputs listed in the settings (the Thinkpad USB-C ones)
![inputoutput](https://user-images.githubusercontent.com/1605980/108500795-9a17e580-72b0-11eb-84d1-4561dd4276cd.png)
The other outputs (speakers-built-in audio and HDMI outputs) are working OK.
Toggling some settings to show/hide items does not change anything.

attached some logs when the extension is started
[log.txt](https://github.com/kgshank/gse-sound-output-device-chooser/files/6009722/log.txt)

list of audio devices:
```
$ arecord -l                                                                                                                                                                                 


**** List of CAPTURE Hardware Devices ****
card 0: PCH [HDA Intel PCH], device 0: CX8070 Analog [CX8070 Analog]
  Subdevices: 0/1
  Subdevice #0: subdevice #0
card 2: Au [ThinkPad USB-C Dock Gen2 USB Au], device 0: USB Audio [USB Audio]
  Subdevices: 0/1
  Subdevice #0: subdevice #0
```

**Comment by mouse256** [comment]
I just noticed that after a reboot everything was fine. But then after resuming from standby the docking was gone again.

**Comment by bmcfee** [comment]
I've also been seeing this problem when docked and resuming from standby.  When waking while connected to a dock, the extension gets stuck in "Extension initialising...".  If I then disconnect and reconnect, everything works again.

**Comment by mouse256** [comment]
> I've also been seeing this problem when docked and resuming from standby. When waking while connected to a dock, the extension gets stuck in "Extension initialising...". If I then disconnect and reconnect, everything works again.

Ah indeed, reconnecting the dock fixes the issue. Good to know.
I don't understand how people use gnome with multiple outputs without this extension. It's really a hell to select the right output device without...

**Comment by bmcfee** [comment]
> I don't understand how people use gnome with multiple outputs without this extension.

My temporary fix is to use the caffeine extension to prevent sleep :shrug: hopefully it gets properly resolved soon though!

**Comment by kgshank** [comment]
I am rewriting the extension completely to avoid memory issues caused by mishandling of C pointers. I will release a working version soon. Please bear with the delay!!

**Comment by mouse256** [comment]
Ok cool. Your work is much appreciated!

**Comment by kgshank** [comment]
Git is now updated with the latest code, please retest with the git version, if possible and let me know if the issue exists

**Comment by bmcfee** [comment]
Thanks @kgshank !  I just gave it a spin and it seems like it's working now.  For the record, here's what I tried:

- upgrade/install extension
- restart gnome (had to reboot for kernel reasons anyway, should be irrelevant here)
- select devices
- suspend machine
- wake machine

Upon wake, all devices showed up again as expected.  Previously, there would just be the "initialising" message.

I'm not sure if this fixes OP's issue, but it at least resolved mine.
